QuoridorEngine/src/ui/gui/GuiClient.cs

```diff
@@ -0,0 +1,361 @@
+            hidePossiblePlayerMoves();
+
+            // Update player pawn location in gui based on last move
+            guiFrame.MovePlayerPawn(isWhitePlayerTurn, playerCell.Row, playerCell.Column);
+
+            // Player's turn has finished
+            switchPlayerTurn();
+        }
+
+        /// <summary>
+        /// Event handler for player pawn click 
+        /// </summary>
+        /// <param name="sender">Clicked player pawn</param>
+        /// <param name="e">Event arguments</param>
+        private void PlayerPawnClicked(object sender, EventArgs e)
+        {
+            bool isWhitePlayer = ((PlayerPawn)sender).IsWhite;
+
+            // Ignore if not player's turn
+            if (isWhitePlayer != isWhitePlayerTurn) return;
+
+            if (initPlayerMove)
+                // Cancel initiated player move
+                hidePossiblePlayerMoves();
+            else
+                // Initiate player move 
+                showPossiblePlayerMoves(isWhitePlayer);
+        }
+
+        /// <summary>
+        /// Update gui board player cells to show player's pawn possible moves
+        /// </summary>
+        /// <param name="isWhitePlayer">Player to show possible moves</param>
+        private void showPossiblePlayerMoves(bool isWhitePlayer)
+        {
+            List<QuoridorMove> possiblePlayerMoves = (List<QuoridorMove>)game.GetPossiblePlayerMoves(isWhitePlayer);
+
+            foreach (QuoridorMove move in possiblePlayerMoves)
+            {
+                (int guiRow, int guiColumn) = TransformCoordinates.GameToGuiPlayer(move.Row, move.Column);
+                guiFrame.PossibleMovePlayerCell(guiRow, guiColumn);
+            }
+
+            // Player pawn move initiated
+            initPlayerMove = true;
+        }
+
+        // Reset all possible player move cells to normal state 
+        private void hidePossiblePlayerMoves()
+        {
+            for (int gameRow = game.Dimension - 1; gameRow >= 0; gameRow--)
+                for (int gameColumn = 0; gameColumn < game.Dimension; gameColumn++)
+                {
+                    (int guiRow, int guiColumn) = TransformCoordinates.GameToGuiPlayer(gameRow, gameColumn);
+                    guiFrame.NormalPlayerCell(guiRow, guiColumn);
+                }
+
+            // Player pawn move completed, canceled or interrupted
+            initPlayerMove = false;
+        }
+
+        /// <summary>
+        /// Utility to refresh/update wall cells on gui based on changes made in quoridor core
+        /// </summary>
+        /// <param name="function"> An action to perform on gui wall cells</param>
+        /// <param name="hasWallPiece">Utility parameter to determine whether a wall piece 
+        /// is placed or not in quoridor core according to performed action</param>
+        private void refreshWallCells(BoardCellAction function, bool hasWallPiece)
+        {
+            for (int gameRow = game.Dimension - 1; gameRow >= 0; gameRow--)
```

Comment:
Run different loops for horizontal and vertical walls since they do different fnctionality for each type and even run up to different boundary values